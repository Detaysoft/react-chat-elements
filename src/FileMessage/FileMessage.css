.rce-mbox-file {
    padding-bottom: 13px;
}

.rce-mbox-file > button {
    background: #e9e9e9;
    display: flex;
    border-radius: 5px;
    margin-top: -3px;
    margin-right: -6px;
    margin-left: -6px;
    align-items: center;
    min-height: 52px;
    max-width: 500px;
    padding: 5px 0;
    cursor: pointer;
    user-select: none;
    outline: none;
    border:none;
}

.rce-mbox-file > button > * {
    padding: 0px 10px;
}

.rce-mbox-file--icon {
    font-size: 30px;
    align-items: center;
    display: flex;
    flex-direction: column;
}

.rce-mbox-file--size {
    font-size: 10px;
    margin-top: 3px;
    max-width: 52px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.rce-mbox-file--text {
    font-size: 13.6px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.rce-mbox-file--buttons {
    font-size: 30px;
    align-items: center;
    display: flex;
}

.rce-mbox-file--loading {
    font-size: 15px;
    width: 40px;
    height: 40px;
}
