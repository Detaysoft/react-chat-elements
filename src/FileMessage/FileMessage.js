import React, { Component } from 'react';
import './FileMessage.css';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';
import FaFile from 'react-icons/lib/fa/file';

const ProgressBar = require('react-progress-bar.js');
const Circle = ProgressBar.Circle;

export class FileMessage extends Component {

    onClick(e) {
        if (!this.props.data.status)
            return;

        if (!this.props.data.status.download && this.props.onDownload instanceof Function)
            this.props.onDownload(e);
        else if (this.props.data.status.download && this.props.onOpen instanceof Function)
            this.props.onOpen(e);
    }

    render() {
        var progressOptions = {
            strokeWidth: 5,
            color: '#333',
            trailColor: '#aaa',
            trailWidth: 5,
            step: (state, circle) => {
                circle.path.setAttribute('trail', state.color);
                circle.path.setAttribute('trailwidth-width', state.width);

                var value = Math.round(circle.value() * 100);
                if (value === 0)
                    circle.setText('');
                else
                    circle.setText(value);
            }
        };

        return (
            <div className='rce-mbox-file'>
                <button onClick={this.onClick.bind(this)}>
                    <div className="rce-mbox-file--icon">
                        <FaFile
                            color='#aaa'/>
                        <div className="rce-mbox-file--size">
                            {this.props.data.size}
                        </div>
                    </div>
                    <div className="rce-mbox-file--text">
                        {this.props.text}
                    </div>
                    <div className="rce-mbox-file--buttons">
                        {
                            this.props.data.status &&
                            !this.props.data.status.download &&
                            !this.props.data.status.click &&
                            <FaCloudDownload
                                color='#aaa'/>
                        }
                        {
                            this.props.data.status &&
                            typeof this.props.data.status.loading === 'number' &&
                            this.props.data.status.loading !== 0 &&
                            <Circle
                                progress={this.props.data.status.loading}
                                options={progressOptions}
                                initialAnimate={true}
                                containerClassName={'rce-mbox-file--loading'} />
                        }
                    </div>
                </button>
            </div>
        );
    }
}

FileMessage.defaultProps = {
    text: '',
    data: {},
    onClick: null,
    onDownload: null,
    onOpen: null,
};


export default FileMessage;
