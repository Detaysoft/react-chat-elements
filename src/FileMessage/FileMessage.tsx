import React from 'react'
import { FaFile, FaCloudDownloadAlt, FaExclamationTriangle } from 'react-icons/fa'
import ProgressCircle from '../Circle/Circle'
import { IFileMessageProps } from '../type'
import './FileMessage.css'

const FileMessage: React.FC<IFileMessageProps> = props => {
  var progressOptions = {
    strokeWidth: 5,
    color: '#333',
    trailColor: '#aaa',
    trailWidth: 5,
    step: (
      state: { color: string; width: string },
      circle: {
        path: { setAttribute: (arg0: string, arg1: string) => void }
        value: () => number
        setText: (arg0: string | number) => void
      }
    ) => {
      circle.path.setAttribute('trail', state.color)
      circle.path.setAttribute('trailwidth-width', state.width)

      var value = Math.round(circle.value() * 100)
      if (value === 0) circle.setText('')
      else circle.setText(value)
    },
  }

  const error = props?.data?.status && props?.data?.status.error === true

  const onClick = (e: React.MouseEvent) => {
    if (!props?.data?.status) return

    if (!props?.data?.status.download && props.onDownload instanceof Function) props.onDownload(e)
    else if (props?.data?.status.download && props.onOpen instanceof Function) props.onOpen(e)
  }

  return (
    <div className='rce-mbox-file'>
      <button onClick={onClick}>
        <div className='rce-mbox-file--icon'>
          <FaFile color='#aaa' />
          <div className='rce-mbox-file--size'>{props?.data.size}</div>
        </div>
        <div className='rce-mbox-file--text'>{props?.data?.name ?? props?.text}</div>
        <div className='rce-mbox-file--buttons'>
          {error && (
            <span className='rce-error-button'>
              <FaExclamationTriangle color='#ff3d3d' />
            </span>
          )}
          {!error && props?.data?.status && !props?.data?.status.download && !props?.data?.status.click && (
            <FaCloudDownloadAlt color='#aaa' />
          )}
          {!error &&
            props?.data?.status &&
            typeof props?.data?.status.loading === 'number' &&
            props?.data?.status.loading !== 0 && (
              <ProgressCircle
                animate={props?.data?.status.loading}
                className='rce-mbox-file--loading'
                progressOptions={progressOptions}
              />
            )}
        </div>
      </button>
      {props?.text && <div className='rce-mbox-text'>{props.text}</div>}
    </div>
  )
}

export default FileMessage
