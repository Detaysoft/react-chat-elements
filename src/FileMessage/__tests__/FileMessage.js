import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FileMessage from '../FileMessage'

describe('FileMessage component', () => {
  it('should render without issues', () => {
    const component = shallow(<FileMessage data={{}} />)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
