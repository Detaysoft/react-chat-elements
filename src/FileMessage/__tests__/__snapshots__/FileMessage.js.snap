// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileMessage component should render without issues 1`] = `
<div
  className="rce-mbox-file"
>
  <button
    onClick={[Function]}
  >
    <div
      className="rce-mbox-file--icon"
    >
      <FaFile
        color="#aaa"
      />
      <div
        className="rce-mbox-file--size"
      />
    </div>
    <div
      className="rce-mbox-file--text"
    />
    <div
      className="rce-mbox-file--buttons"
    />
  </button>
</div>
`;
